appcues/src/main/java/com/appcues/ui/presentation/EmbedViewPresenter.kt

```diff
@@ -0,0 +1,28 @@
+package com.appcues.ui.presentation
+
+import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import com.appcues.RenderContextManager
+import com.appcues.data.model.RenderContext
+import org.koin.core.scope.Scope
+
+internal class EmbedViewPresenter(
+    scope: Scope,
+    renderContext: RenderContext,
+    private val renderContextManager: RenderContextManager,
+) : ViewPresenter(scope, renderContext) {
+
+    override fun ViewGroup.setupView(): ComposeView? {
+        val embedView = renderContextManager.getEmbedView(renderContext) ?: return null
+
+        return ComposeView(context).also {
+            embedView.addView(it)
+        }
+    }
+
+    override fun ViewGroup.removeView() {
+        post {
+            renderContextManager.getEmbedView(renderContext)?.removeAllViews()
```

Comment:
is there any chance that other views might exist in the container and we should only remove our own? I'm guessing no, but checking.

Reply:
Maybe it would be better off creating our own ViewGroup child like AppcuesEmbeddedContent: FrameLayout